sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala

```diff
@@ -72,9 +72,8 @@ class QueryCompilationErrorsSuite extends QueryTest with SharedSparkSession {
 
       Dataset.ofRows(spark, plan)
     }.message
-    assert(msg.contains("The feature is not supported: " +
+    assert(msg.matches("The feature is not supported: " +
```

Comment:
is this a user-facing error? The test constructs logical plan directly which is not user-facing.

Reply:
The exception was introduced by https://github.com/apache/spark/pull/28572. @Ngone51 Can it be triggered by user code?

Probably, it would be nice to revise all existing error classes/exception, and check their visibility to users.

Reply:
Opened SPARK-38131 for the task.